reframe/core/decorators.py

```diff
@@ -91,6 +91,16 @@ def _validate_test(cls):
         raise ValueError(f'decorated test ({cls.__qualname__!r}) has one or '
                          f'more undefined parameters')
 
+    conditions = [VersionValidator(v) for v in cls._rfm_required_version]
+    if (cls._rfm_required_version and
+        not any(c.validate(osext.reframe_version()) for c in conditions)):
+
+        getlogger().info('skipping incompatible test defined'
+                         ' in class: %s' % cls.__name__)
```

Comment:
I think we need a more explicit message here such as `"test 'MyTest' is incompatible with the current ReFrame version (x.x.x)"`